NewForce-Cohort-9/tabloidmvc-stafford-infection-1
# Delete Posts Feature

# Description

Added functionality for users to delete their own posts. Users can now navigate to a post they have authored, view a confirmation page, and confirm the deletion of the post. Also ensured that only the author of a post can delete it.

Addresses Trello Ticket #7, "Delete A Post" https://trello.com/c/ZRnA1rjE/7-7-delete-a-post

## Type of change




# Testing Instructions

1. Clone down the repository
2. git checkout main
3. git fetch --all
4. git checkout tyler-delete-posts-1
5. If you don't see the most recent version of my files, git pull origin tyler-delete-posts-1
6. Open the project (TabloidMVC) in Visual Studio
7. If you haven't already, create a database called "TabloidMVC."  The data we are using can be found under the "SQL" folder when looking at our group repository.  Be sure to run a query for both files (Create_DB and Seed_Data)
8. Start the server.  It should automatically open a webpage on your browser. 
9. You'll be greeted by a page that will say "Blogs & Journals", "Get Started", and "Login."  Click on Login
10. Login using admin@example.com and hit "Submit"
11. Using the navbar at the top, navigate to "Posts"
12. Click on "New Posts"
13. Fill out all of the boxes (Title, Header Image URL, Published, Category, and Content) with whatever information you like.  You don't have to include a functioning URL to test with.
14. Click "Save" to create a new post
15. You should see a page for your new post.
16. EITHER click the blue trashcan "Delete" on your post page OR navigate back to posts and click the delete icon next to your new post.
17. You should be redirected to a page with all of your post information asking you whether you're sure you want to delete your post or not.
18. Click "Back to List" and verify it returns you to the list of posts.
19. Click on the trashcan icon by your post again.  You can either do this from the list of posts, or from the details page about your post.
20. Click the red "Delete" button.  You will be navigated to the list of posts, and the post you deleted will be gone.
21. Congratulations, you have deleted a post.



# Checklist:
